### ISSUE TITLE
[false-negative]ICAST_BAD_SHIFT_AMOUNT can not detect int literal shift

### ISSUE THREAD

**Issue by wuchiuwong** [spotbugs/spotbugs#1143]
**Affects SpotBugs Version:** 
spotbugs-4.0.0-RC2

**Rule:**
ICAST_BAD_SHIFT_AMOUNT

**Description:**
Unable to detect int literal shift, for example: <code>1 << 48</code>.
This rule is implemented in SpotBugs.jar\main\java\edu\umd\cs\findbugs\detect\FindPuzzlers.java,
```
if (leftHandSide instanceof Integer && ((Integer) leftHandSide) > 0) {
	// boring; lie so other detectors won't get concerned
	constantArgumentToShift = true;
	valueOfConstantArgumentToShift = 8;
}
```
It seems that this rule deliberately exclude the constant shift, but we think it is also worth checking

**Code Sample demonstrating the issue:**
```
if (cost < 0 || cost > (1 << 48))
cost = 1 << 48;
long posMask = ~(1 << 63);
```

**Expected outcome:**
false-negative

**Running SpotBugs through:**
command line
